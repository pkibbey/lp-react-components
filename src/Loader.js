import React from 'react';
import { Image } from 'rebass';
import Spinner from './assets/oval.svg';

const Loader = () => (
  <Image alt="" data-testid="loader" src={Spinner} height={24} width={24} />
);

export default Loader;
