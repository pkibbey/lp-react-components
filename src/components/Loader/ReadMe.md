Default loader

```js
<Loader variant="dark" />
```

Small loader

```js
<Loader variant="dark" size="small" />
```

Large loader

```js
<Loader variant="dark" size="large" />
```
