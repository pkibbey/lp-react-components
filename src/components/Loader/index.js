import React from 'react'
import PropTypes from 'prop-types'
import theme from '../../theme'
import Spinner from 'react-spinner-material'

const convertSizetoPx = (size) => {
  switch (size) {
    case 'small':
      return 18
    case 'large':
      return 40
    default:
      return 24
  }
}

/**
 * Used to convey to the user that something is happening, usually an asyncronous network call
 */
const Loader = ({ size, variant, timeout }) => (
  <Spinner
    data-testid='loader'
    type='Oval'
    radius={convertSizetoPx(size)}
    stroke={2}
    color={variant === 'dark' ? theme.colors.darkGray : 'white'}
  />
)

Loader.propTypes = {
  /** The size of the loader */
  size: PropTypes.oneOf(['small', 'large', 'default']),
  /** A variant for the loader */
  variant: PropTypes.oneOf(['light', 'dark']),
  /** A time in ms that the loader should run for ('undefined' repeats it forever) */
  timeout: PropTypes.number
}

Loader.defaultProps = {
  size: 'default',
  variant: 'light',
  timeout: undefined
}

export default Loader
