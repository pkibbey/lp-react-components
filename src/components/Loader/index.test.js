import React from 'react'
import { render } from '@testing-library/react'
import Loader from './'

it('renders a default loader', () => {
  const { container } = render(<Loader />)
  const element = container.querySelector('.react-spinner-material')
  expect(element).toBeEmptyDOMElement()
  expect(element).toHaveStyle({
    width: '24px',
    height: '24px',
    borderColor: 'white',
    borderWidth: '2px'
  })
})

it('renders a small loader', () => {
  const { container } = render(<Loader size='small' />)
  const element = container.querySelector('.react-spinner-material')
  expect(element).toBeEmptyDOMElement()
  expect(element).toHaveStyle({
    width: '18px',
    height: '18px',
    borderColor: 'white',
    borderWidth: '2px'
  })
})

it('renders a large loader', () => {
  const { container } = render(<Loader size='large' />)
  const element = container.querySelector('.react-spinner-material')
  expect(element).toBeEmptyDOMElement()
  expect(element).toHaveStyle({
    width: '40px',
    height: '40px',
    borderColor: 'white',
    borderWidth: '2px'
  })
})
