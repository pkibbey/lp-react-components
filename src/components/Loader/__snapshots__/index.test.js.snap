// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a default loader 1`] = `
<div
  className="react-spinner-material"
  style={
    Object {
      "borderColor": "white",
      "borderWidth": 2,
      "height": 24,
      "width": 24,
    }
  }
/>
`;

exports[`renders a large loader 1`] = `
<div
  className="react-spinner-material"
  style={
    Object {
      "borderColor": "white",
      "borderWidth": 2,
      "height": 40,
      "width": 40,
    }
  }
/>
`;

exports[`renders a small loader 1`] = `
<div
  className="react-spinner-material"
  style={
    Object {
      "borderColor": "white",
      "borderWidth": 2,
      "height": 18,
      "width": 18,
    }
  }
/>
`;
